test/fixtures/controller.line/borderCapStyle/scriptable.js

```diff
@@ -0,0 +1,64 @@
+module.exports = {
+	config: {
+		type: 'line',
+		data: {
+			labels: [0, 1, 2, 3, 4, 5],
+			datasets: [
+				{
+					// option in dataset
+					data: [4, 5, 10, null, -10, -5],
+					borderCapStyle: function(ctx) {
+						var dataTotal = ctx.dataset.data.reduce(function(a, b) {
+							return a + b;
+						});
+						var style = dataTotal > 0 ? 'round' : 'square';
+
+						return style;
+					}
+				},
+				{
+					// option in element (fallback)
+					data: [-4, -5, -10, null, 10, 5],
+				}
+			]
+		},
+		options: {
+			legend: false,
+			title: false,
+			elements: {
```

Comment:
The main issue with these border cap tests is that we barely see the `borderCapStyle` in the PNG files, meaning that it will probably not fail if something change. We should make sure that the line is not cut by adding padding `layout: {padding: ?}`, but also certainly increase the `borderWidth` (maybe 24 or 32?)

Reply:
@simonbrunel Thank you for this review! I will get to work on these changes.

Reply:
<a href="/chartjs/Chart.js/pull/6128/commits/fd338d7a0e48039b3b1cd219d64d18474ba48c2b" class="commit-id">fd338d7</a> . I added padding and borderWidth and did what I could to improve the clarity of the tests.

Reply:
Looks really great. I would just try to remove the background gray color/gradient (not sure from where it comes).